QuasarHelios2015/src/org/team1540/quasarhelios/Elevator.java

```diff
@@ -29,20 +33,18 @@
         }
     }
 
-    private static final BooleanStatus raising = new BooleanStatus();
-    private static final BooleanStatus lowering = new BooleanStatus();
-    private static final BooleanStatus goingToMiddle = new BooleanStatus();
+    private static final BooleanStatus raising = new BooleanStatus(false);
+    private static final BooleanStatus lowering = new BooleanStatus(false);
 
-    public static final EventOutput setTop = EventMixing.combine(raising.getSetTrueEvent(), lowering.getSetFalseEvent(), goingToMiddle.getSetFalseEvent());
-    public static final EventOutput setMiddle = goingToMiddle.getSetTrueEvent();
-    public static final EventOutput setBottom = EventMixing.combine(raising.getSetFalseEvent(), lowering.getSetTrueEvent(), goingToMiddle.getSetFalseEvent());
-    public static final EventOutput stop = BooleanMixing.getSetEvent(BooleanMixing.combine(raising, lowering, goingToMiddle), false);
+    public static final BooleanStatus overrideEnabled = new BooleanStatus(false);
+    public static final FloatStatus overrideValue = new FloatStatus(0.0f);
 
-    public static final BooleanInput topLimitSwitch = BooleanMixing.createDispatch(BooleanMixing.invert(Igneous.makeDigitalInput(0)), Igneous.globalPeriodic);
-    public static final BooleanInput bottomLimitSwitch = BooleanMixing.createDispatch(BooleanMixing.invert(Igneous.makeDigitalInput(1)), Igneous.globalPeriodic);
-    public static final BooleanInput middleUpperLimitSwitch = BooleanMixing.createDispatch(BooleanMixing.invert(Igneous.makeDigitalInput(6)), Igneous.globalPeriodic);
-    public static final BooleanInput middleLowerLimitSwitch = BooleanMixing.createDispatch(BooleanMixing.invert(Igneous.makeDigitalInput(7)), Igneous.globalPeriodic);
-    private static BooleanStatus lastLimitSide = new BooleanStatus(); // true = top, false = bottom
+    public static final EventOutput setTop = EventMixing.combine(raising.getSetTrueEvent(), lowering.getSetFalseEvent());
+    public static final EventOutput setBottom = EventMixing.combine(raising.getSetFalseEvent(), lowering.getSetTrueEvent());
+    public static final EventOutput stop = BooleanMixing.getSetEvent(BooleanMixing.combine(raising, lowering), false);
+
+    public static final BooleanStatus atTop = new BooleanStatus(false);
+    public static final BooleanStatus atBottom = new BooleanStatus(true);
```

Comment:
Why do we assume that it's at the bottom initially? Also, since the public part here is a BooleanStatus, this means that other classes can modify the value. Is that intended?